compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -289,185 +274,24 @@ class ExtractSemanticDB extends Phase:
-
-      def addOwner(owner: Symbol): Unit =
-        if !owner.isRoot then addSymName(b, owner)
-
-      def addOverloadIdx(sym: Symbol): Unit =
-        val decls =
-          val decls0 = sym.owner.info.decls.lookupAll(sym.name)
-          if sym.owner.isAllOf(JavaModule) then
-            decls0 ++ sym.owner.companionClass.info.decls.lookupAll(sym.name)
-          else
-            decls0
-        end decls
-        val alts = decls.filter(_.isOneOf(Method | Mutable)).toList.reverse
-        def find(filter: Symbol => Boolean) = alts match
-          case notSym :: rest if !filter(notSym) =>
-            val idx = rest.indexWhere(filter).ensuring(_ >= 0)
-            b.append('+').append(idx + 1)
-          case _ =>
-        end find
-        val sig = sym.signature
-        find(_.signature == sig)
-
-      def addDescriptor(sym: Symbol): Unit =
-        if sym.is(ModuleClass) then
-          addDescriptor(sym.sourceModule)
-        else if sym.is(TypeParam) then
-          b.append('['); addName(sym.name); b.append(']')
-        else if sym.is(Param) then
-          b.append('('); addName(sym.name); b.append(')')
-        else if sym.isRoot then
-          b.append(Symbols.RootPackage)
-        else if sym.isEmptyPackage then
-          b.append(Symbols.EmptyPackage)
-        else if (sym.isScala2PackageObject) then
-          b.append(Symbols.PackageObjectDescriptor)
-        else
-          addName(sym.name)
-          if sym.is(Package) then b.append('/')
-          else if sym.isType || sym.isAllOf(JavaModule) then b.append('#')
-          else if sym.isOneOf(Method | Mutable)
-          && (!sym.is(StableRealizable) || sym.isConstructor) then
-            b.append('('); addOverloadIdx(sym); b.append(").")
-          else b.append('.')
-
-      /** The index of local symbol `sym`. Symbols with the same name and
-       *  the same starting position have the same index.
-       */
-      def localIdx(sym: Symbol)(using Context): Int =
-        val startPos =
-          assert(sym.span.exists, s"$sym should have a span")
-          sym.span.start
-        @tailrec
-        def computeLocalIdx(sym: Symbol): Int = locals get sym match
-          case Some(idx) => idx
-          case None      => symsAtOffset(startPos).find(_.name == sym.name) match
-            case Some(other) => computeLocalIdx(other)
-            case None =>
-              val idx = nextLocalIdx
-              nextLocalIdx += 1
-              locals(sym) = idx
-              symsAtOffset(startPos) += sym
-              idx
-        end computeLocalIdx
-        computeLocalIdx(sym)
-      end localIdx
-
-      if sym.exists then
-        if sym.isGlobal then
-          addOwner(sym.owner); addDescriptor(sym)
-        else
-          b.append(Symbols.LocalPrefix).append(localIdx(sym))
-
-    end addSymName
-
-    /** The semanticdb name of the given symbol */
-    private def symbolName(sym: Symbol)(using Context): String =
```

Comment:
Just moved to `SemanticSymbolBuilder.scala` because we want to use this method outside from `ExtractSemanticDB.scala`